pyiceberg/expressions/parser.py

```diff
@@ -217,12 +219,25 @@ def _(result: ParseResults) -> BooleanExpression:
 
 @starts_with.set_parse_action
 def _(result: ParseResults) -> BooleanExpression:
-    return StartsWith(result.column, result.raw_quoted_string)
+    return _evaluate_like_statement(result)
 
 
 @not_starts_with.set_parse_action
 def _(result: ParseResults) -> BooleanExpression:
-    return NotStartsWith(result.column, result.raw_quoted_string)
+    return _evaluate_like_statement(result).__invert__()
```

Comment:
The `__function__` of not used explicitly in Python. For example, instead of `__str__()`, it is more pythonic to use `str(obj)`.

```suggestion
    return ~_evaluate_like_statement(result)
```